src/wallet/wallet.cpp

```diff
@@ -1085,11 +1105,13 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransaction& tx, const uint256& bl
             }
 
             CWalletTx wtx(this, tx);
-
-            if (saplingNoteData && !saplingNoteData->empty()) {
-                wtx.SetSaplingNoteData(*saplingNoteData);
+            if (!saplingNoteDataMap.empty()) {
+                wtx.SetSaplingNoteData(saplingNoteDataMap);
             }
 
+            // Add external notes info if we are sending
+            if (isFromMe) AddExternalNotesDataToTx(wtx);
```

Comment:
Why not adding this to `saplingNoteDataMap` one line below calling `FindNotesDataAndAddMissingIVKToKeystore` and re using the ` wtx.SetSaplingNoteData` to set them all at once

Reply:
...meh..